app/server.js

```diff
@@ -1,13 +1,329 @@
+let brands = [];
+let products = [];
+let users = [];
+const accessTokens = [
+  {
+    username: 'yellowleopard753',
+    lastUpdated: new Date(),
+    token: 'oo5DD2jLOTLR9s5t'
+  },
+  {
+    username: 'lazywolf342',
+    lastUpdated: new Date(),
+    token: 'SjlNLkhNnA7DuCJb'
+  }
+];
 
-http.createServer(function (request, response) {
+// Set up router
+const myRouter = Router();
+myRouter.use(bodyParser.json());
 
-}).listen(PORT);
\ No newline at end of file
+// Set up server
+const server = http
+  .createServer((request, response) => {
+    myRouter(request, response, finalHandler(request, response));
+  })
+  .listen(PORT, (error) => {
+    if (error) {
+      return console.log('Error on Server Startup: ', error);
+    }
+
+    // Load data
+    fs.readFile(
+      path.join(__dirname, '..', 'initial-data', 'brands.json'),
+      (error, data) => {
+        if (error) throw error;
+        brands = JSON.parse(data);
+        console.log(`Server setup: ${brands.length} brands loaded`);
+      }
+    );
+
+    fs.readFile(
+      path.join(__dirname, '..', 'initial-data', 'products.json'),
+      (error, data) => {
+        if (error) throw error;
+        products = JSON.parse(data);
+        console.log(`Server setup: ${products.length} products loaded`);
+      }
+    );
+
+    fs.readFile(
+      path.join(__dirname, '..', 'initial-data', 'users.json'),
+      (error, data) => {
+        if (error) throw error;
+        users = JSON.parse(data);
+        console.log(`Server setup: ${users.length} users loaded`);
+      }
+    );
+  });
+
+// Set up /api/brands endpoint handler
+myRouter.get('/api/brands', (request, response) => {
+  // Return all brands
+  response.writeHead(200, { 'Content-Type': 'application/json' });
+  return response.end(JSON.stringify(brands));
+});
+
+// Set up /api/brands/{brandId}/products endpoint handler
+myRouter.get('/api/brands/:brandId/products', (request, response) => {
+  const { brandId } = request.params;
+
+  // If brand ID parameter is invalid, return 400 error
+  if (!isPositiveInteger(brandId)) {
+    return return400Error(response);
+  }
+
+  // If brand ID does not exist, return 404 error
+  if (!idExists(brandId, brands)) {
+    return return404Error(response);
+  }
+
+  // Otherwise, return all products for a given brand
+  const brandProducts = products.filter(
+    (product) => product.categoryId === brandId
+  );
+
+  response.writeHead(200, { 'Content-Type': 'application/json' });
+  return response.end(JSON.stringify(brandProducts));
+});
+
+// Set up /api/products endpoint handler
+myRouter.get('/api/products', (request, response) => {
+  const queryParams = queryString.parse(url.parse(request.url).query);
+
+  // If search term provided, return only products that match
+  if (queryParams.query) {
+    const matchingProducts = products.filter(
+      (product) =>
+        product.name.toLowerCase().includes(queryParams.query.toLowerCase()) ||
```

Comment:
style wise this is a little messy, could make it easier to read by putting all the includes on their own line